client/csharp/Script Wars Client/Script Wars Client/ScriptWars/Games/Tanks/World.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using ScriptWars.Connection;
+
+namespace ScriptWars.Games.Tanks
+{
+    /// <summary>
+    /// The world map.
+    /// </summary>
+    public class World
+    {
+        /// <summary>
+        /// The width of the map in squares.
+        /// </summary>
+        public int Width => _map.GetLength(0);
+
+        /// <summary>
+        /// The height of the map in squares.
+        /// </summary>
+        public int Height => _map.GetLength(1);
+
+        /// <summary>
+        /// Positions of all ammo packs on the map.
+        /// </summary>
+        public ICollection<Coordinates> AmmoLocations { get; }
+
+        /// <summary>
+        /// All <see cref="Tank"/> visible to the client.
+        /// </summary>
+        public ICollection<Tank> Tanks { get; }
+
+        /// <summary>
+        /// All <see cref="Shot"/> visible to the client.
+        /// </summary>
+        public ICollection<Shot> Shots { get; }
+
+        private readonly bool[,] _map;
+        private readonly IDictionary<Coordinates, Tank> _tankMap = new Dictionary<Coordinates, Tank>();
+
+        /// <summary>
+        /// Read the game world from the server tick response.
+        /// </summary>
+        /// <param name="network"><see cref="Network"/> to use to fetch data.</param>
+        public World(Network network)
+        {
+            // Map dimensions
+            int width = network.ReadByte();
+            int height = network.ReadByte();
+
+            _map = new bool[width, height];
+
+            // Wall positions
+            for (var i = 0; i < _map.GetLength(0); i++)
+            for (var j = 0; j < _map.GetLength(1); j++)
+                _map[i, j] = network.ReadBool();
```

Comment:
Yeah, I'll test that again. I thought I was reading this incorrectly and that was causing the coordinate issues.